tasks/main.yml

```diff
@@ -0,0 +1,127 @@
+---
+- name: Gather needed facts
+  setup:
+    gather_subset:
+      - '!all'
+      - '!min'
+      - distribution_major_version
+  when: ansible_facts.distribution_major_version is not defined
+
+- name: System check
+  fail:
+    msg: >-
+      Only Red Hat Enterprise Linux >= 8.1 is supported;
+      System: {{ ansible_facts.os_family }}
+      Version: {{ ansible_facts.distribution_version }}
+  when: (ansible_facts.os_family != "RedHat") or
+    (ansible_facts.distribution_version is version("8.1", "<"))
+
+- name: Check trust compatibility
+  fail:
+    msg: >-
+      Fapolicyd does not support trust setting fapolicyd_setup_trust
+      on EL version < 8.3
+  ignore_errors: true
+  when:
+    - fapolicyd_setup_trust | length > 0
+    - ansible_facts.distribution_version is version("8.2", "<=")
+  register: __failed_check_trust
+
+- name: Check integrity compatibility
+  fail:
+    msg: >-
+      Fapolicyd does not support integrity setting fapolicyd_setup_integrity
+      on EL version < 8.4
+  ignore_errors: true
+  when:
+    - fapolicyd_setup_integrity | length > 0
+    - ansible_facts.distribution_version is version("8.3", "<=")
+  register: __failed_check_integrity
+
+- name: Check trust files compatibility
+  fail:
+    msg: >-
+      Fapolicyd does not support trust files setting fapolicyd_add_trusted_file
+      on EL version < 8.4
+  ignore_errors: true
+  when:
+    - fapolicyd_add_trusted_file | length > 0
+    - ansible_facts.distribution_version is version("8.3", "<=")
```

Comment:
```suggestion
    - ansible_distribution in ['CentOS', 'RedHat']
    - ansible_distribution_version is version("8.3", "<=")
```

Reply:
Same as above.

Reply:
We don't need to check this because correct distro/version is checked in system check.